shopinvader_payment/services/abstract_payment.py

```diff
@@ -94,62 +121,119 @@ def _validator_check_payment(self):
         return validator
 
     # Private method
-    def _set_payment_mode(self, cart, params):
-        payment_mode_id = params['payment_mode']['id']
+    def _set_payment_mode(self, target, params):
+        """
+
+        :param target: payment recordset
+        :param params: dict
+        :return: str
+        """
+        payment_mode_id = params.get('payment_mode', {}).get('id')
         available_payment_mode_ids = [
-            x['id'] for x in self._get_available_payment_mode(cart)]
+            x.get('id') for x in self._get_available_payment_mode(target)]
         if payment_mode_id not in available_payment_mode_ids:
             raise UserError(_('Unsupported payment mode'))
         else:
-            vals = cart.play_onchanges({
+            vals = target.play_onchanges({
```

Comment:
@acsonefho Strange: at line 57  https://github.com/akretion/odoo-shopinvader/pull/233/files#diff-75a89e68649ad3cf5b9b50cdd683df00R57 you check if `'payment_mode_id` is a field on target and here you play the onchange for this file without checking if it exists on `target`